core/events/events_selected.ts

```diff
@@ -68,11 +69,16 @@ export class Selected extends UiBase {
    *
    * @param json JSON representation.
    */
-  override fromJson(json: AnyDuringMigration) {
+  override fromJson(json: SelectedJson) {
     super.fromJson(json);
     this.oldElementId = json['oldElementId'];
     this.newElementId = json['newElementId'];
   }
 }
 
+export interface SelectedJson extends AbstractEventJson {
+  oldElementId: string|null;
```

Comment:
similarly, most of the other event types seem to use `undefined` for missing properties, why do these use `null`?

Reply:
The constructor accepts null, so unless we want to change that, these also have to accept null :/

Personally I think it would be a good idea to make all of the events consistently be `SomeType|undefined` but I didn't want to just do that without discussing.

Reply:
Aside: I prefer `null` to `undefined` as the 'empty' value of normally-object-valued properties and variables.  This is probably mostly just because I'm a (now slightly literally) greybeard C hacker, but there is a practical justification as well: since `undefined` in JS can also mean 'uninitialised' one can't be sure a value was actually set, whereas `null` shows that it was definitely initialised but is empty.  So encountering `undefined` unexpectedly tells you something has gone wrong.

It's also the convention used for [[Prototype]]: you can have a `null`-prototype object, but not an `undefined`-prototype one.

(I think the Closure type system designers must have felt similarly, too, because I note that it has a very concise syntax for indicating nullability, but no special syntax for defining undefinedability.)

Reply:
> (I think the Closure type system designers must have felt similarly, too, because I note that it has a very concise syntax for indicating nullability, but no special syntax for defining undefinedability.)

As opposed to typescript, which has the opposite (special syntax for undefinablility, and no special syntax for nullability).

Reply:
> As opposed to typescript, which has the opposite (special syntax for undefinablility, and no special syntax for nullability).

True.  And fail to make a distinction between properties that do not exist and properties which exist but which have the value `undefined`.  😢